### PR TITLE
Add octree stuff

### PR DESCRIPTION
**Pull request by cphyc**
This adds the section in "abstracting simulation types" relevant to oct-based datasets as well as a comment to prevent us from forgetting to discuss oct-vs-patch ray traversal for volume rendering.

Any comment is welcome!

### CODE REVIEW

File: content/30.abstracting_simulation_types.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ In the limit of many more cells than grid objects, this overhead is small, but i
 These cases are becoming more common even for medium-scale simulations.
 
 To address both the memory overhead and the python overhead, as well as more generally address potential scalability issues with grid selection, we have begun implementation of a more sophisticated "grid visitors" indexing and selection method.
+**CC: this should not be marked as "ongoing" anymore, should it?**
```
Comment (matthewturk) [inline content/30.abstracting_simulation_types.md:47]: "It is still ongoing!" he said optimistically and self-consciously.

File: content/30.abstracting_simulation_types.md [hunk 2]
```diff
@@ -54,6 +55,35 @@ The spatial-tree optimization substantially increases performance for "wide and
 
 ### Octree Analysis {#sec:octree_analysis}
 
+`yt` supports octree-based AMR datasets (RAMSES, Art).
```
Comment (matthewturk) [inline content/30.abstracting_simulation_types.md:58]:
```suggestion
`yt` supports octree-based AMR datasets (primarily RAMSES and ART, but also the output from the octree-based radiative transfer code Hyperion).
```

File: content/30.abstracting_simulation_types.md [hunk 3]
```diff
@@ -54,6 +55,35 @@ The spatial-tree optimization substantially increases performance for "wide and
...
+
+In order to load data within a given region, a two-step approach is followed.
+First, the cells interesting with the region of interest, as described in [Data Objects](#sec:data_objects) are selected. `yt` relies on an oct-visitor machinery combined with selection routines.
+The tree is recursively traversed depth-first starting from the root grid, following only those branches that may intersect with the selected region.
+At the tip of each branch, the up-to-eight leaf cells are visited.
+In a first pass, the number of selected cells is computed and in a second pass, the on-file location of their parent oct is stored.
+Second, `yt` relies on the on-file location obtained from the octree traversal to lazily read data from disk.
+This ensures that only the minimal amount of data is being read and is particularly efficient when accessing a region spanning a small number of domains and/or a small number of refinement levels.
+
+`yt` also fully supports accessing neighboring cells.
```
Comment (matthewturk) [inline content/30.abstracting_simulation_types.md:71]:
```suggestion
Recently, `yt` has been extended to fully supports accessing neighboring cells.
```

File: content/30.abstracting_simulation_types.md [hunk 4]
```diff
@@ -54,6 +55,35 @@ The spatial-tree optimization substantially increases performance for "wide and
...
+This is achieved by computing one-cell thick quantities around each oct, which emulates the `ghost zones` found in patch-based codes. This approach has the advantage of abstracting the octree structure and provides a common interface to create derived fields, as described in **CC: any section describing this?**.
```
Comment (matthewturk) [inline content/30.abstracting_simulation_types.md:72]:
```suggestion
This is achieved by computing one-cell thick quantities around each oct, which emulates the "ghost zones" found in patch-based codes. This approach has the advantage of abstracting the octree structure and provides a common interface to create derived fields, as described in **CC: any section describing this?**.
```

Not sure yet about the section describing it, so let's leave that bold comment in and I'll come back to it.

File: content/30.abstracting_simulation_types.md [hunk 5]
```diff
@@ -54,6 +55,35 @@ The spatial-tree optimization substantially increases performance for "wide and
...
+The 56 neighbors ($4^3 - 2^3$) surrounding each oct are found by performing a search in the octree, which finds any neighbor in $O(\mathrm{level})$, where $\mathrm{level}$ is the level of the central oct.
+The search is illustrated on Figure {@fig:binary-search}.
+**CC: remove this? Instead of starting the search from the root grid, it could be possible to start from the central oct and go upwards. This has not been implemented yet.**
```
Comment (matthewturk) [inline content/30.abstracting_simulation_types.md:75]: I think it's okay to leave in if it's how we do it, and leave a note that we think other paths are possible.

**Review by matthewturk — APPROVED** [review]
This is really awesome stuff.

**Comment by cphyc** [comment]
Note that I added two pdf illustrations, but they look a bit funny on the html version of the paper. I would rather not convert them to png (they are 100% vectorized, so that would be a shame). @matthewturk any suggestion (assuming we keep them)?

**Comment by matthewturk** [comment]
@cphyc Were the figures made using TikZ?  If so, do you have the TikZ script?  If you check that in under `scripts` I'd be happy to add something to convert them into one or more formats for the paper.

**Comment by cphyc** [comment]
Well spotted! The binary search thingy was generated with TikZ and the following gist https://gist.github.com/cphyc/c3603a564eb9d1ea743c3ea9d5f0f9df (note that there is an illustration of a binary search in 1D as well if that is of interest). The other figure for the gradient was made with Inkscape (with the tex+pdf export mode).

**Pull request merged**